tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -419,6 +419,11 @@ class CompilerTest {
   TEST_METHOD(CodeGenDot1)
   TEST_METHOD(CodeGenDynamic_Resources)
   TEST_METHOD(CodeGenEffectSkip)
+  TEST_METHOD(CodeGenEliminateDynamicIndexing)
+  TEST_METHOD(CodeGenEliminateDynamicIndexing2)
+  TEST_METHOD(CodeGenEliminateDynamicIndexing3)
+  TEST_METHOD(CodeGenEliminateDynamicIndexing4)
+  TEST_METHOD(CodeGenEliminateDynamicIndexing5)
```

Comment:
Lets add tests to exercise the multiple return code path. Also would be good to have a case that has mixed dynamic/non-dynamic stores.